src/lib/pass_one/pass_one.rs

```diff
@@ -41,43 +49,151 @@ fn get_instruction_size (inst:&Instruction) -> i32 {
             match operands[0].val {
                 Value::Raw(x) => return x as i32,
                 Value::SignedInt(x) => return x,
-                _ => panic!("Unexpected Error"),
+                _ => panic!("Unexpected Error {:?}", *inst),
             }
-        },
+        }
         "RESW" => {
             let operands = unwrap_to_vec(&inst.operands);
             if operands.len() != 1 {
                 panic!("RESW expects only 1 operand");
             }
-             match operands[0].val {
-                 Value::Raw(x) => return x as i32,
-                 Value::SignedInt(x) => return x,
-                 _ => panic!("Unexpected Error"),
-             }
-        },
+            match operands[0].val {
+                Value::Raw(x) => return x as i32,
+                Value::SignedInt(x) => return x,
+                _ => panic!("Unexpected Error"),
+            }
+        }
         _ => (),
     }
     return 0;
 }
-pub fn pass_one (start: &i32, mut file:FileHandler) ->(HashMap<String, i32>, Vec<Instruction>) {
-    let mut loc  = *start;
-    let mut symbol_table:HashMap<String, i32> = HashMap::new();
-    let mut listing:Vec<Instruction> = Vec::new();
-    let mut instruction;
-    loop {
-        match file.read_instruction() {
-            None => break,
-            Some(inst) => instruction = inst,
-        }
+
+pub fn pass_one(mut file: FileHandler) -> (HashMap<String, i32>, RawProgram) {
+
```

Comment:
no they don't, it's a convention to leave empty lines in code and between functions